src/hooks/useGoalResource.ts

```diff
@@ -1,49 +1,446 @@
+                },
+            })();
+        },
+        [convertGoalCriteria, invalidate],
+    );
+
+    const onGoalStateChange = useCallback(
+        async (nextState: GoalStateChangeSchema['state'], invalidateKey?: RefetchKeys | RefetchKeys[]) => {
+            if (!id) return;
+
+            await stateChangeMutations.mutateAsync({
+                id,
+                state: nextState,
+            });
+
+            invalidate(invalidateKey);
+        },
+        [id, invalidate, stateChangeMutations],
+    );
+
+    const onGoalDelete = useCallback(
+        async (invalidateKey?: RefetchKeys | RefetchKeys[]) => {
+            if (!id) return;
+
+            const [data] = await notifyPromise(
+                archiveMutation.mutateAsync({
+                    id,
+                    archived: true,
+                }),
+                'goalsDelete',
+            );
+
+            if (data) {
+                invalidate(invalidateKey);
+            }
+        },
+        [id, archiveMutation, invalidate],
+    );
+
+    const onGoalParticipantAdd = useCallback(
+        async (activityId?: string | null, invalidateKey?: RefetchKeys | RefetchKeys[]) => {
+            if (!id || !activityId) return;
+
+            await addParticipantMutation.mutateAsync({ id, activityId });
+
+            invalidate(invalidateKey);
+        },
+        [addParticipantMutation, id, invalidate],
+    );
+    const onGoalParticipantRemove = useCallback(
+        (activityId?: string | null, invalidateKey?: RefetchKeys | RefetchKeys[]) => async () => {
+            if (!id || !activityId) return;
+
+            await removeParticipantMutation.mutateAsync({ id, activityId });
+
+            invalidate(invalidateKey);
+        },
+        [id, invalidate, removeParticipantMutation],
+    );
+
+    const onChangeGoalProject = useCallback(
+        async (projectId: string) => {
+            if (!projectId || !id) return;
+
+            const promise = changeProjectMutation.mutateAsync({
+                id,
+                projectId,
+            });
+
+            await notifyPromise(promise, 'goalsUpdate');
+
+            return promise;
+        },
+        [changeProjectMutation, id],
     );
 
     return {
         toggleGoalWatching,
         toggleGoalStar,
+        onUpdateGoal,
+        onGoalStateChange,
+        onGoalDelete,
+        onChangeGoalProject,
+
+        helpers: {
+            invalidate,
+            registerArgs,
+        },
+
+        participants: {
+            add: onGoalParticipantAdd,
+            remove: onGoalParticipantRemove,
+        },
+
+        dependency: {
+            add: onAddGoalDependency,
+            remove: onRemoveGoalDependency,
+        },
+
+        draft: {
+            saveCommentDraft,
+            resolveCommentDraft,
+            removeCommentDraft,
+        },
+
+        comment: {
+            highlightCommentId,
+            lastStateComment,
+
+            change: onCommentChange,
```

Comment:
U mustn't rename handlers? What the reason? Your code can be shorter if you don't do that.